piptools/scripts/compile.py

```diff
@@ -155,13 +154,32 @@ def cli(
     config: Path | None,
     no_config: bool,
     constraint: tuple[str, ...],
+    build_deps_for_distributions: tuple[Literal["sdist", "wheel", "editable"], ...],
+    all_build_deps: bool,
+    build_deps_only: bool,
 ) -> None:
     """
     Compiles requirements.txt from requirements.in, pyproject.toml, setup.cfg,
     or setup.py specs.
     """
     log.verbosity = verbose - quiet
 
+    if all_build_deps and build_deps_for_distributions:
+        raise click.BadParameter(
+            "--build-deps-for has no effect when used with --all-build-deps"
+        )
+    elif all_build_deps:
+        build_deps_for_distributions = ALL_BUILD_DISTRIBUTIONS
```

Comment:
```suggestion
        build_deps_for_distributions = options.ALL_BUILD_DISTRIBUTIONS
```

We can use imported options module.

Reply:
Will fix.